output/cloud/expv2/metrics_client.go

```diff
@@ -108,5 +104,10 @@ func newRequestBody(data *pbcloud.MetricSet) ([]byte, error) {
 		return nil, fmt.Errorf("the Protobuf message is too large to be handled by Snappy encoder; "+
 			"size: %d, limit: %d", len(b), 0xffffffff)
 	}
-	return snappy.Encode(nil, b), nil
+	encoded := snappy.Encode(nil, b)
+	if len(b) > payloadSizeLimit {
```

Comment:
Just double-checking if we should check the original payload size, not the compressed (encoded) one right?

Reply:
FYI, nop based on the fact that the compression making a lot of reduction it was better to do it after otherwise, we could have it this quite often. Anyway, this is now not relevant anymore.